Providers/Modules/Plugins/Antimalware/plugin/collectmcafeeinfo.rb

```diff
@@ -9,34 +9,40 @@
 
 class McAfee
 
-	def self.detect()	
-		begin
-			if !File.file?('/opt/isec/ens/threatprevention/bin/isecav')
-				return false
-			end			
-			detectioncmd = `/opt/isec/ens/threatprevention/bin/isecav --version 2>&1`.lines.map(&:chomp)
+	attr_accessor :detectedPath , :mcafeeName , :mcafeeVersion 
+    def self.findMcAfeePath()
+        paths = ['/opt/McAfee/ens/tp/bin/mfetpcli','/opt/isec/ens/threatprevention/bin/isecav']   
+        for path in paths 
+            if File.file?(path)
+                @detectedPath = path
+                detectioncmd = `#{path} --version 2>&1`.lines.map(&:chomp)
+				@mcafeeName = detectioncmd[0]
+                @mcafeeVersion = detectioncmd[1].split(" : ")[1]
```

Comment:
Consider breaking the loop if detection is successful for a path

Reply:
sure will post this in next revision.

Reply:
Added break statement